app/connectors/CustomsDeclareExportsMovementsConnector.scala

```diff
@@ -31,47 +32,42 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 @Singleton
-class CustomsDeclareExportsMovementsConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient) {
+class CustomsDeclareExportsMovementsConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient)(
+  implicit ec: ExecutionContext
+) {
 
   private val logger = Logger(this.getClass)
 
   private val CustomsDeclareExportsMovementsUrl = s"${appConfig.customsDeclareExportsMovements}"
 
   private val movementSubmissionUrl: PartialFunction[ActionType, String] = {
-    case Arrival            => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementArrivalSubmissionUri}"
-    case Departure          => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementDepartureSubmissionUri}"
-    case DucrAssociation    => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementConsolidationAssociateUri}"
-    case DucrDisassociation => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementConsolidationDisassociateUri}"
-    case ShutMucr           => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementConsolidationShutMucrUri}"
+    case Arrival   => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementArrivalSubmissionUri}"
+    case Departure => s"$CustomsDeclareExportsMovementsUrl${appConfig.movementDepartureSubmissionUri}"
+    case DucrAssociation | DucrDisassociation | ShutMucr =>
+      s"$CustomsDeclareExportsMovementsUrl${appConfig.movementConsolidationUri}"
   }
 
   private val CommonMovementsHeaders =
     Seq(HeaderNames.CONTENT_TYPE -> ContentTypes.XML(Codec.utf_8), HeaderNames.ACCEPT -> ContentTypes.XML(Codec.utf_8))
 
-  def sendArrivalDeclaration(
-    requestXml: String
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = postRequest(Arrival, requestXml)
+  private val JsonHeaders = Seq(HeaderNames.CONTENT_TYPE -> ContentTypes.JSON, HeaderNames.ACCEPT -> ContentTypes.JSON)
 
-  def sendDepartureDeclaration(
-    requestXml: String
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = postRequest(Departure, requestXml)
+  def sendArrivalDeclaration(requestXml: String)(implicit hc: HeaderCarrier): Future[HttpResponse] =
+    postRequest(Arrival, requestXml)
 
-  def sendAssociationRequest(
-    requestXml: String
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = postRequest(DucrAssociation, requestXml)
+  def sendDepartureDeclaration(requestXml: String)(implicit hc: HeaderCarrier): Future[HttpResponse] =
+    postRequest(Departure, requestXml)
 
-  def sendDisassociationRequest(
-    requestXml: String
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
-    postRequest(DucrDisassociation, requestXml)
+  def sendConsolidationRequest(consolidation: ConsolidationRequest)(implicit hc: HeaderCarrier): Future[HttpResponse] =
+    httpClient.POST[ConsolidationRequest, HttpResponse](
```

Comment:
return type `HttpResponse`?

Reply:
Means its leaking Connector code to the Service

Reply:
I'd just return ConsolidationRequest, but call it `Consolidation` or something

Reply:
Just for now I returned HttpResponse, because our backend doesn't return ConsolidationRequest actually, but Consolidation, another class that we need to build in the frontend to be consistent. As I discussed with you we will return Request from the backend but I don't want to break actual implementation. I can prepare parallel PR for movements to return ConsoldationRequest and change it here. It'll be better or you have different proposal?